lib/active_record/connection_adapters/crate_adapter.rb

```diff
@@ -131,11 +131,16 @@ def connect
 
       def columns(table_name) #:nodoc:
         cols = @connection.table_structure(table_name).map do |field|
-          CrateColumn.new(field[2], nil, field[4], nil)
+          name = dotted_name(field[2])
+          CrateColumn.new(name, nil, field[4], nil)
         end
         cols
       end
 
+      def dotted_name(name)
```

Comment:
no idea how complicated that is do achieve with ruby, otherwise an integration test with different possible column names